push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/NotificationDispatcher.java

```diff
@@ -86,13 +100,21 @@ public SenderServiceCallback(Variant variant, int tokenSize, FlatPushMessageInfo
 
         @Override
         public void onSuccess() {
-            logger.debug(String.format("Sent '%s' message to '%d' devices", variant.getType().getTypeName(), tokenSize));
+            // add to a Kafka topic that one more message was sent successfully
+            pushMessageSendingMetricsProducer.send(KAFKA_PUSH_MESSAGE_SENDING_METRICS_TOPIC, KAFKA_PUSH_MESSAGE_SENDING_SUCCESS);
+            logger.debug("Sent {} message to {} devices", variant.getType().getTypeName(), tokenSize);
         }
 
         @Override
         public void onError(final String reason) {
-            logger.warn(String.format("Error on '%s' delivery: %s", variant.getType().getTypeName(), reason));
+            logger.warn("Error on {} delivery: {}", variant.getType().getTypeName(), reason);
             pushMessageMetricsService.appendError(pushMessageInformation, variant, reason);
+            // add to a Kafka topic that a message was sent unsuccessfully
+            pushMessageSendingMetricsProducer.send(KAFKA_PUSH_MESSAGE_SENDING_METRICS_TOPIC, KAFKA_PUSH_MESSAGE_SENDING_FAILURE);
```

Comment:
would it make sense to use the jobId as the key ? (e.g. `.getId` on the `FlatPMI` ?)